add ARD function for table_02

ayogasekaram opened the pull request:
Initial commit
- create the ARD function, make table generation functions internal.

jknizia commented on R/fda-table_02.R:
```diff
@@ -1,24 +1,20 @@
-#' * Numbers in table for non-numeric variables represent the absolute numbers of patients and fraction of `N`.
-#' * All-zero rows are removed by default (see `prune_0` argument).
 #'
+#' @inheritParams tbl_make_table_02
```
This link does not exist. Please update to `make_table_02`

jknizia replied:
Also for `@seealso`

jknizia reviewed (changes requested):
Hi @ayogasekaram, thanks for adding me to the PR. I found a couple of minor issues. Please see comments in the code. BR

ayogasekaram commented:
Thank you so much for your review @jknizia! Much appreciated :) 
I believe I've addressed all of the comments but if there's anything else that needs to be fixed - please let me know.

jknizia commented on quarto/table-templates/template-table_02.qmd:
```diff
@@ -28,29 +31,23 @@ adsl <- adsl %>%
-anl <- left_join(adsl, advs, by = "USUBJID")
 
 # Output Table
 make_table_02(
```
Not sure whether this is only me, but... If I run this example, no table is produced. If I instead assign the make_table_02() call to a variable and then call the variable, the table is produced. Could you please double check this behavior from your end?

jknizia replied:
It works now, thanks!

jknizia commented on quarto/table-templates/template-table_02.qmd:
```diff
@@ -59,35 +56,35 @@ make_table_02(
+| **Argument** | **Description** | **Default** |
+| `df` | (`data.frame`) Dataset (typically ADSL) required to build table. | *No default* |
+| `return_ard` | (`flag`) Whether an ARD should be returned. | `TRUE` |
+| `denominator` | (`character`) Alternative dataset used only to calculate column counts. | `NULL` |
```
Please remove `denominator` entry from this table

jknizia commented on quarto/table-templates/template-table_02.qmd:
```diff
@@ -174,42 +143,51 @@ advs <- advs %>%
+| **Argument** | **Description** | **Default** |
+|:-----------------------|:-----------------------|:-----------------------|
+| `df` | (`data.frame`) Dataset (typically ADSL) required to build table. | *No default* |
+| `alt_counts_df` | (`character`) Alternative dataset used only to calculate column counts. | `NULL` |
```
Please remove `alt_counts_df` entry

jknizia commented on quarto/table-templates/template-table_02.qmd:
```diff
@@ -174,42 +143,51 @@ advs <- advs %>%
+| `vars` | (`character`) Variables from `df` to include in the table. | `c("SEX", "AGE", "AGEGR1", "RACE", "ETHNIC", "COUNTRY")` |
+| `lbl_vars` | (`character`) Labels corresponding to variables in `vars` to print in the table. Labels should be ordered according to the order of variables in `vars`. | `formatters::var_labels(df, fill = TRUE)[vars]` |
+| `lbl_overall` | (`character`) If specified, an overall column will be added to the table with the given value as the column label. | `"Total Population"` |
+| `.stats` | (`character`) Statistics to include in the table. Includes statistics for all variable types (only the statistics that are valid for a given variable's type will be printed). See [`tern::analyze_vars()`](https://insightsengineering.github.io/tern/main/reference/analyze_variables.html#value) for options. | `c("mean_sd", "median_range", "count_fraction")` |
```
Please remove `.stats` entry, as it is not used within the rtable producing function

jknizia commented on quarto/table-templates/template-table_02.qmd:
```diff
@@ -174,42 +143,51 @@ advs <- advs %>%
+| `lbl_vars` | (`character`) Labels corresponding to variables in `vars` to print in the table. Labels should be ordered according to the order of variables in `vars`. | `formatters::var_labels(df, fill = TRUE)[vars]` |
+| `lbl_overall` | (`character`) If specified, an overall column will be added to the table with the given value as the column label. | `"Total Population"` |
+| `.stats` | (`character`) Statistics to include in the table. Includes statistics for all variable types (only the statistics that are valid for a given variable's type will be printed). See [`tern::analyze_vars()`](https://insightsengineering.github.io/tern/main/reference/analyze_variables.html#value) for options. | `c("mean_sd", "median_range", "count_fraction")` |
+| `.formats` | (named `list` of `character`) List of formats corresponding to each value in `.stats`. Each name is a value in `.stats` and the corresponding value is the format that should be applied to that statistic. See [`formatters::list_valid_format_labels()`](https://insightsengineering.github.io/formatters/main/reference/list_valid_format_labels.html#ref-examples) for a list of valid forma ...
```
Please remove `.formats` entry, as it is not used within the rtable producing function

jknizia reviewed (changes requested):
Dear @ayogasekaram, thank you so much for the update. I found only a few other notes :)

edelarua commented on R/fda-table_02.R:
```diff
@@ -29,274 +25,127 @@ NULL
+    gtsummary::modify_footnote(everything() ~ NA) %>%
+    gtsummary::modify_column_alignment(columns = all_stat_cols(), align = "right")
+
+  if (return_ard) {
+    ard <- gtsummary::gather_ard(tbl)
+    res <- list(tbl = tbl, ard = ard)
+  } else {
+    res <- list(tbl = tbl)
   }
```
This will fix the no output that Jessica noticed.
```suggestion
  if (return_ard) {
    ard <- gtsummary::gather_ard(tbl)
    res <- list(tbl = tbl, ard = ard)
  } else {
    res <- list(tbl = tbl)
  }
```

edelarua commented on quarto/table-templates/template-table_02.qmd:
```diff
@@ -28,29 +31,23 @@ adsl <- adsl %>%
     AGE >= 75 ~ ">=75"
   )))
 
```
No label for new variable
```suggestion
attr(adsl$AGEGR1, "label") <- "Age Group"

```

edelarua reviewed (commented):
Hi @ayogasekaram,

A few additional comments from me but it looks great!!

jknizia commented on quarto/table-templates/template-table_02.qmd:
```diff
@@ -27,30 +30,24 @@ adsl <- adsl %>%
-  )
+  df = adsl,
+  vars = c("SEX", "AGE", "AGEGR1", "RACE", "ETHNIC", "COUNTRY"),
+  return_ard = FALSE
```
It seems like the gtsummary table example produces not quite the same table as the rtables table example. There is an additional parameter "Baseline Temperature" displayed in the rtables example. Could you align the two examples, please?

ayogasekaram replied:
great call, Jessica! I've added the AVAL to all examples since the demographics table should analyze some baseline characteristics as well :)

jknizia reviewed (approved):
Hi @ayogasekaram, thanks for the updates. I added one further recommendation, but since it is not essential for the functions to work, I approve already.
